test-harness/modules/mocker/common.py

```diff
@@ -0,0 +1,119 @@
+# -*- coding: utf-8 -*-
+# Copyright (C) Mesosphere, Inc. See LICENSE file for details.
+
+"""
+Shared management code for DC/OS mocks used by AR instances, both EE and Open.
+"""
+
+import logging
+
+from mocker.endpoints.reflectors import (
+    ReflectingTcpIpEndpoint,
+    ReflectingUnixSocketEndpoint,
+)
+from mocker.endpoints.mesos import MesosEndpoint
+from mocker.endpoints.marathon import MarathonEndpoint
+
+log = logging.getLogger(__name__)
+
+
+class MockerBase():
+    """This class represents mocking behaviour shared between both EE and Open
+       repositories.
+
+    It should not be instantiated directly but instead inheriting classes should
+    override/extend it's methods.
+    """
+    _endpoints = None
+
+    def _register_endpoints(self, endpoints):
+        """Register given endpoints list with the mock
+
+        This method registers all the endpoints that are going to be managed
+        by this Mocker instance.
+
+        Args:
+            endpoints (object: [EndpointA, EndpointB,...]): list of endpoints
+                that should be registered
+        """
+        self._endpoints = {}
+        for endpoint in endpoints:
+            log.info("Registering endpoint `%s`", endpoint.id)
+            assert endpoint.id not in self._endpoints
+            self._endpoints[endpoint.id] = endpoint
+
+    @staticmethod
+    def _create_common_endpoints():
+        """Helper function that takes care of creating/instantiating all the
+           endpoints that are common for both EE and Open repositories"""
+        res = []
+
+        # pkgpanda endpoint
+        res.append(ReflectingUnixSocketEndpoint('/run/dcos/pkgpanda-api.sock'))
+        # exhibitor
+        res.append(ReflectingTcpIpEndpoint(ip='127.0.0.1', port=8181))
+        # mesos
+        res.append(MesosEndpoint(ip='127.0.0.1', port=5050))
+        # marathon
+        res.append(MarathonEndpoint(ip='127.0.0.1', port=8080))
+        # TODO - other endpoints go here...
+
+        return res
+
+    def __init__(self, extra_endpoints=None):
+        """Initialize new MockerBase instance
+
+        Args:
+            extra_endpoints (obj: [EndpointA, EndpointB,...]): list of endpoints
+                that are unique to the inheriting class/represent specific behaviour
+                of given flavour
+        """
+        common_endpoints = self._create_common_endpoints()
+        endpoints = common_endpoints + extra_endpoints
+        self._register_endpoints(endpoints)
+
+    def start(self):
+        """Start all endpoints registered with this Mocker instance"""
+        for endpoint in self._endpoints.values():
+            endpoint.start()
+
+    def stop(self):
+        """Stop all endpoints registered with this Mocker instance.
+
+        Usually called right before object destruction
+        """
+        for endpoint in self._endpoints.values():
```

Comment:
Pawel: Not at all necessary - these endpoints are unrelated except for the fact that they share a common logger. This common logger is cleanup up after they are stopped.